clab/config.go

```diff
@@ -270,30 +270,49 @@ func (c *CLab) processStartupConfig(nodeCfg *types.NodeConfig) error {
 		return err
 	}
 
-	// if the startup-config is a remote file, download it to the temp directory
-	if strings.HasPrefix(p, "http://") || strings.HasPrefix(p, "https://") {
-		tmpLoc := c.TopoPaths.ClabTmpDir()
+	// if more then 1 newline, then it is an embedded config
+	isEmbeddedConfig := strings.Count(p, "\n") > 1
+	// if it starts with http(s):// it is a downloadable config
+	isDownloadableConfig := strings.HasPrefix(p, "http://") || strings.HasPrefix(p, "https://")
+
+	if isEmbeddedConfig || isDownloadableConfig {
 
+		// if the startup-config is a remote file, download it to the temp directory
+		tmpLoc := c.TopoPaths.ClabTmpDir()
 		utils.CreateDirectory(tmpLoc, 0755)
 
-		// get file name from an URL
-		fname := utils.FilenameForURL(p)
+		switch {
+		case isEmbeddedConfig:
+			// we have an embedded config with multiple newlines
+			// its an embedded config
+			absDestFile := c.TopoPaths.StartupConfigDownloadFileAbsPath(nodeCfg.ShortName, "embedded.config")
```

Comment:
I think this must be named `embedded.partial.cfg` because full configs are not good for embedding, and we have to treat them as partial

Reply:
they are not good ... but they work.
So your opinion is to have partial configs provided inline?

Reply:
yes, the idea was to support embedding partials